tests/internals/polling_cooldown_so/polling_cooldown_so_test.go

```diff
@@ -0,0 +1,309 @@
+metadata:
+  name: {{.TriggerAuthName}}
+  namespace: {{.TestNamespace}}
+spec:
+  secretTargetRef:
+    - parameter: username
+      name: {{.SecretName}}
+      key: AUTH_USERNAME
+    - parameter: password
+      name: {{.SecretName}}
+      key: AUTH_PASSWORD
+`
+
+	deploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{.DeploymentName}}
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+spec:
+  selector:
+    matchLabels:
+      app: {{.DeploymentName}}
+  replicas: 0
+  template:
+    metadata:
+      labels:
+        app: {{.DeploymentName}}
+    spec:
+      containers:
+      - name: nginx
+        image: nginxinc/nginx-unprivileged
+        ports:
+        - containerPort: 80
+`
+
+	metricsServerDeploymentTemplate = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.MetricsServerDeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.MetricsServerDeploymentName}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{.MetricsServerDeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.MetricsServerDeploymentName}}
+    spec:
+      containers:
+      - name: metrics
+        image: ghcr.io/kedacore/tests-metrics-api
+        ports:
+        - containerPort: 8080
+        envFrom:
+        - secretRef:
+            name: {{.SecretName}}
+        imagePullPolicy: Always
+`
+
+	scaledObjectTemplate = `
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{.ScaledObject}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  scaleTargetRef:
+    name: {{.DeploymentName}}
+  pollingInterval: {{.PollingInterval}}
+  cooldownPeriod: {{.CooldownPeriod}}
+  minReplicaCount: {{.MinReplicas}}
+  maxReplicaCount: {{.MaxReplicas}}
+  triggers:
+  - type: metrics-api
+    metadata:
+      targetValue: "1"
+      url: "{{.MetricsServerEndpoint}}"
+      valueLocation: 'value'
+      method: "query"
+    authenticationRef:
+      name: {{.TriggerAuthName}}
+`
+
+	updateMetricsTemplate = `
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: update-ms-value-
+  namespace: {{.TestNamespace}}
+spec:
+  ttlSecondsAfterFinished: 0
+  template:
+    spec:
+      containers:
+      - name: job-curl
+        image: curlimages/curl
+        imagePullPolicy: Always
+        command: ["curl", "-X", "POST", "{{.MetricsServerEndpoint}}/{{.MetricValue}}"]
+      restartPolicy: Never
+`
+
+	serviceTemplate = `
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{.ServiceName}}
+  namespace: {{.TestNamespace}}
+spec:
+  selector:
+    app: {{.MetricsServerDeploymentName}}
+  ports:
+  - port: 8080
+    targetPort: 8080
+`
+)
+
+// use KubectlCreateWithTemplate() function because applying yaml template
+// for update-metrics relies on the previous job to be deleted
+// (job is immutable), but in this case the test runs too fast at some
+// points that job is not finished & deleted in time
```

Comment:
Isn't`KubectlApplyWithTemplate` with `ttlSecondsAfterFinished: 0` enough for removing the job?

Reply:
its not, the job has a few seconds to run and it didnt finish in time.